parser_library/include/protocol.h

```diff
@@ -115,6 +117,38 @@ struct PARSER_LIBRARY_EXPORT completion_item
 template class PARSER_LIBRARY_EXPORT sequence<completion_item, const lsp::completion_item_s*>;
 using completion_list = sequence<completion_item, const lsp::completion_item_s*>;
 
+enum class PARSER_LIBRARY_EXPORT document_symbol_kind
+{
+    DAT = 0,
+    EQU = 1,
+    MACH = 2,
+    UNKNOWN = 3,
+    VAR = 4,
+    SEQ = 5,
+    COMMON = 6,
+    DUMMY = 7,
+    EXECUTABLE = 8,
+    READONLY = 9,
+    MACRO = 10
+};
+
+struct PARSER_LIBRARY_EXPORT document_symbol_item;
+template class PARSER_LIBRARY_EXPORT sequence<document_symbol_item, const lsp::document_symbol_item_s*>;
+using document_symbol_list = sequence<document_symbol_item, const lsp::document_symbol_item_s*>;
+
+struct PARSER_LIBRARY_EXPORT document_symbol_item
+{
+    document_symbol_item(const lsp::document_symbol_item_s& item);
+    context::id_index name() const;
```

Comment:
this line needs the #include "context/id_storage.h"  @slavek-kucera 
I could change the id_index to string and then the dependency would not be needed.
Should I do it?

Reply:
`context::id_index` is internal and should not be used in `protocol.h` - `sequence<char>` should probably be used instead.